### PR TITLE
Enable frozen string literals

### PR DESCRIPTION
**Pull request by tagliala**
Blocked by:
- [x] #171
- [x] #173
- [x] #174
- [x] #175

### CODE REVIEW

File: spec/io_resource_spec.rb [hunk 1]
```diff
@@ -60,20 +62,20 @@
     context 'IO object' do
       let(:answer) { 'read' }
       let(:rio) { StringIO.new(answer, 'r') }
-      let(:wio) { StringIO.new('write', 'w') }
+      let(:wio) { StringIO.new(+'write', 'w') }
```
Comment (tagliala) [inline spec/io_resource_spec.rb:65]: this is the only noticeable change in this PR. I've searched for potential issues like `<< "` or `= ''` but I couldn't find them

**Comment by tagliala** [comment]
Hi @jamesmartin, sorry for the ping

if you can take a look at 

- #173 (trivial to review)
- #174 (trivial to review, there is also a RuboCop cop for this now)
- #175 (3 commits, 1 automatic safe fix by rubocop, 2 changes easy to review)

Then we can go for this one. I'm also working on profiling and benchmarks

**Comment by jamesmartin** [comment]
@tagliala thanks for the ping. I've reviewed, approved and merged the three PRs on which this branch depends. Feel free to push this one forward whenever you have time. 👍

**Comment by tagliala** [comment]
@jamesmartin thanks, pushed

I cannot guarante 100% that there isn't an edge case which breaks with frozen string literals, but at the moment I didn't find one.

Here there are some comparisons on memory side. I don't know if this makes a difference when a page contains a single `inline_svg_tag`

### Bench
```rb
def run_benchmark_sample(iterations: 1)
  # Plain
  HELPER.inline_svg_tag 'example.svg'

  # Class
  HELPER.inline_svg_tag 'example.svg', class: 'test-class'

  # Id
  HELPER.inline_svg_tag 'example.svg', id: 'test-id'

  # Aria
  HELPER.inline_svg_tag 'example.svg', title: 'Title', desc: 'Description', aria: true

  # Style
  HELPER.inline_svg_tag 'example.svg', style: 'margin: 0;'

  # ...
  HELPER.inline_svg_tag 'example.svg', class: 'test-class',
                                       id: 'test-id',
                                       style: 'margin: 0;',
                                       title: 'Title', desc: 'Description', aria: true

  # Different SVG
  HELPER.inline_svg_tag 'static_assets/assets0/some-document.svg', class: 'test-class'
end
```

### Memory

#### Before

```
$ ./benchmarks/profile_memory.rb 
Total allocated: 43984 bytes (451 objects)
Total retained:  240 bytes (6 objects)

allocated memory by gem
-----------------------------------
     43984  inline_svg/lib

allocated memory by file
-----------------------------------
     31224  ./lib/inline_svg/transform_pipeline/transformations.rb
      4680  ./lib/inline_svg/transform_pipeline/transformations/transformation.rb
      2080  ./lib/inline_svg/id_generator.rb
      1960  ./lib/inline_svg/transform_pipeline.rb
      1680  ./lib/inline_svg/transform_pipeline/transformations/aria_attributes.rb
       840  ./lib/inline_svg/transform_pipeline/transformations/class_attribute.rb
       480  ./lib/inline_svg/transform_pipeline/transformations/style_attribute.rb
       400  ./lib/inline_svg/transform_pipeline/transformations/description.rb
       400  ./lib/inline_svg/transform_pipeline/transformations/title.rb
       160  ./lib/inline_svg/action_view/helpers.rb
        80  ./lib/inline_svg/transform_pipeline/transformations/id_attribute.rb
```

#### After

```
$ ./benchmarks/profile_memory.rb 
Total allocated: 39984 bytes (351 objects)
Total retained:  80 bytes (2 objects)

allocated memory by gem
-----------------------------------
     39984  inline_svg/lib

allocated memory by file
-----------------------------------
     31224  ./lib/inline_svg/transform_pipeline/transformations.rb
      3120  ./lib/inline_svg/transform_pipeline/transformations/transformation.rb
      1960  ./lib/inline_svg/transform_pipeline.rb
      1920  ./lib/inline_svg/id_generator.rb
       720  ./lib/inline_svg/transform_pipeline/transformations/aria_attributes.rb
       240  ./lib/inline_svg/transform_pipeline/transformations/class_attribute.rb
       240  ./lib/inline_svg/transform_pipeline/transformations/description.rb
       240  ./lib/inline_svg/transform_pipeline/transformations/title.rb
       160  ./lib/inline_svg/action_view/helpers.rb
       160  ./lib/inline_svg/transform_pipeline/transformations/style_attribute.rb
```

**Review by jamesmartin — APPROVED** [review]
@tagliala thanks for the PR and for the benchmarks. ✨  

> I cannot guarante 100% that there isn't an edge case which breaks with frozen string literals, but at the moment I didn't find one.

Let's trust the test suite and move forward from here. 👍
